core/gen_index.ts

```diff
@@ -26,8 +26,9 @@ export function genIndex(base64EncodedConfig: string): string {
     dataform.ProjectConfig.create(config.compileConfig.projectConfigOverride).toJSON()
   );
 
-  const returnValue = !!config.compileConfig.query
-    ? `(function() {
+  const returnValue =
```

Comment:
urgh, i see the issue now, thanks for debugging.

the problem with this fix is that it's reliant on protobufjs mis-implementing the proto3 spec. that empty `query` field _should not be set_, and it's just luck that it is set.

can we instead a boolean field to `CompileConfig` called `compile_single_query` or something, to properly indicate that there is a query to be compiled (even if it's empty)?